File: internal/models/archive.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Archive struct {
	ID          uuid.UUID `json:"id"`
	Name        string    `json:"name"`
	Description string    `json:"description"`
	SourceURL   string    `json:"source_url"`
	Tags        []string  `json:"tags"`
	CreatedAt   time.Time `json:"created_at"`
}

File: front/src/models/archive.ts
export interface Archive {
    id: string;
    name: string;
    filename: string;
    description: string;
    source_url: string;
    tags: string[];
    created_at: string;
    size_bytes: number;
}

export interface GetArchivesResponse {
	archives: Archive[];
	next_cursor: string;
}

export interface GetArchiveTagsResponse {
	tags: string[];
}
